backend/app/models.py

```diff
@@ -413,3 +413,49 @@ def by_transaction(cls, db, txn: Transaction):
             .filter(StripeTransaction.transaction == txn.id)
             .first()
         )
+
+
+class UserOwnedApp(Base):
+    __tablename__ = "userownedapp"
+
+    app_id = Column(String, nullable=False, primary_key=True)
+    account = Column(
+        Integer,
+        ForeignKey(FlathubUser.id, ondelete="CASCADE"),
+        nullable=False,
+        primary_key=True,
+    )
+    created = Column(DateTime, nullable=False)
+
+    @staticmethod
+    def user_owns_app(db, user: FlathubUser, app_id: str):
+        return (
+            db.session.query(UserOwnedApp)
+            .filter_by(account=user.id, app_id=app_id)
+            .first()
+            is not None
+        )
+
+    @staticmethod
+    def all_by_user(db, user: FlathubUser):
+        return db.session.query(UserOwnedApp).filter_by(account=user.id)
+
+    @staticmethod
+    def delete_hash(hasher: utils.Hasher, db, user: FlathubUser):
```

Comment:
I don't think this is `delete`?

Reply:
I'm not sure what you mean. It's added to TABLES_FOR_DELETE so we can remove a user's info when they delete their account. Although, maybe it's worth discussing if we really want to forget app ownership.

Reply:
Nevermind, that code is basically already in place for each model. I don't think I understand what it does, but I'm fine with that.

Reply:
So this is more `mark_for_deletion` then actual `deletion`? That's probably why I was confused by the add.